src/RulesDeploy.ts

```diff
@@ -66,7 +77,39 @@ export class RulesDeploy {
         })
       );
     });
-    return Promise.all(promises);
+    return Promise.all(promises).catch(async (err) => {
+      if (err.status === QUOTA_EXCEEDED_STATUS_CODE) {
+        utils.logBullet(
+          clc.bold.yellow(this.type + ":") + " quota exceeded error while uploading rules"
+        );
+
+        const history: ListRulesetsEntry[] = await gcp.rules.listAllRulesets(this.options.project);
+
+        if (history.length > RULESET_COUNT_LIMIT) {
+          const shouldContinue = await prompt.once({
+            type: "confirm",
+            message: `You have ${
+              history.length
+            } rules, do you want to delete the oldest ${RULESETS_TO_GC} to free up space?`,
+            default: false,
+          });
+          if (shouldContinue) {
+            // Find the oldest unreleased rulesets. The rulesets are sorted reverse-chronlogically.
+            const releases: Release[] = await gcp.rules.listAllReleases(this.options.project);
+            const isReleased = (ruleset: ListRulesetsEntry) =>
+              !!_.find(releases, (release) => release.rulesetName === ruleset.name);
+            const unreleased: ListRulesetsEntry[] = _.reject(history, isReleased);
+            const entriesToDelete = unreleased.reverse().slice(0, RULESETS_TO_GC);
+            for (const entry of entriesToDelete) {
+              await gcp.rules.deleteRuleset(this.options.project, gcp.rules.getRulesetId(entry));
```

Comment:
Maybe add 

```
logger.debug(`[rules] Deleted ruleset ${gcp.rules.getRulesetId(entry)}`)
```

Here so the debug logs will show a bit more progress while waiting for these.

Reply:
done